src/styled/style.tsx

```diff
@@ -32,4 +32,38 @@ const ContentStyle = styled("div", {
   height: "100vh",
 });
 
-export { HeadStyle, ContentStyle };
+type CardStyles = {
+  Root: {
+    display: string;
+    flexDirection: string;
+    flexWrap: string;
+    justifyContent: string;
+    alignItems: string;
+    width: string;
+    minHeight: string;
+    borderRadius: string;
+    margin: string;
+    textAlign: string;
+    backgroundColor: string;
+    border: string;
+  };
+};
+
+const cardStyles: CardStyles = {
+  Root: {
+    display: "flex",
+    flexDirection: "row",
+    flexWrap: "wrap",
+    justifyContent: "center",
+    alignItems: "center",
+    width: "200px",
+    minHeight: "250px",
+    borderRadius: "10px",
+    margin: "40px",
+    textAlign: "center",
+    backgroundColor: "gray",
+    border: "2px solid",
+  },
+};
+
+export { HeadStyle, ContentStyle, cardStyles };
```

Comment:
You need not create multiple objects for each code block, nor do you need to add custom typing like `CardStyles` type.

Refer to this template:


```ts
export const styles: Record<string, React.CSSProperties> = {
  Root: {
    display: "flex",
  },
  Content: {
    ..........
  },
};
```

Here Record is a Typescript utility type that is used to type dynamic objects. It expects 2 arguments, 1st - type of the object's keys, in our case it can be any string like Root, content etc. so `string`, and 2nd arg - the type of the object's values , in our case CSS properties, so we can use the React's builtin `CSSProperties` type.

Reply:
Removes types for cardstyles but utility not working with error: -


`No overload matches this call.
  Overload 1 of 2, '(component: "div", style: StyleObject | ((props: Omit<{}, "$theme"> & { $theme: Theme; }) => StyleObject)): StyletronComponent<"div", {}>', gave the following error.
    Argument of type '{ alignItems: string; backgroundImage: string; backgroundColor: string; backgroundSize: string; backgroundPosition: string; backgroundRepeat: string; boxShadow: string; display: string; flexWrap: string; justifyContent: string; margin: string; width: string; height: string; }' is not assignable to parameter of type 'StyleObject | ((props: Omit<{}, "$theme"> & { $theme: Theme; }) => StyleObject)'.
      Type '{ alignItems: string; backgroundImage: string; backgroundColor: string; backgroundSize: string; backgroundPosition: string; backgroundRepeat: string; boxShadow: string; display: string; flexWrap: string; justifyContent: string; margin: string; width: string; height: string; }' is not assignable to type 'StyleObject'.
        Type '{ alignItems: string; backgroundImage: string; backgroundColor: string; backgroundSize: `

Reply:
Did you try adding this typing template after removing your custom Cardstyles type?

export const styles: Record<string, StyleObject> = {
  Root: {
    display: "flex",
  },
  Content: {
    ..........
  },
};

Reply:
It works now, thanks!